tuning/sft_trainer.py

```diff
@@ -316,13 +316,22 @@ def train(
     # this validation, we just drop the things that aren't part of the SFT Config and build one
     # from our object directly. In the future, we should consider renaming this class and / or
     # not adding things that are not directly used by the trainer instance to it.
+
     transformer_train_arg_fields = [x.name for x in dataclasses.fields(SFTConfig)]
     transformer_kwargs = {
         k: v
         for k, v in train_args.to_dict().items()
         if k in transformer_train_arg_fields
     }
-    training_args = SFTConfig(**transformer_kwargs)
+
+    additional_args = {
+        "dataset_text_field": dataset_text_field,
+    }
+    if is_pretokenized_dataset(
+        data_args.training_data_path or data_args.validation_data_path
+    ):
+        additional_args["dataset_kwargs"] = {"skip_prepare_dataset": True}
+    training_args = SFTConfig(**transformer_kwargs, **additional_args)
```

Comment:
```suggestion

    additional_args = {
        "dataset_text_field": dataset_text_field,
        "dataset_kwargs": dataset_kwargs,
        "max_seq_length": max_seq_length
    }
    training_args = SFTConfig(**transformer_kwargs, **additional_args)
```


@Luka-D Lets go with this then? this should suffice with the recent data processor v1 changes. WDYT?

Reply:
@kmehant Looks good to me. I wasn't the one who added the `if is_pretokenized_dataset(` check, I had just modified it for the new arguments, so I am unsure if it is needed or not. I think the `dataset_kwargs` variable is empty for now, but we can keep it there for future use.

Reply:
@Luka-D I saw that this is merged but can you clarify what you mean that the variable is empty for now?

Reply:
dataset_kwargs would be empty in the sense that probably None in most cases except when using pretokenized datasets where we provide the option to skip the processing of the data to TRL.